ivre/db/mongo.py

```diff
@@ -4478,3 +4504,1129 @@ def get_masters(self):
+        for i, elt in enumerate(group_fields):
+            for field in internal_fields[i]:
+                cur_field = "$%s" % field
+                field_name = None
+                if cur_field in reverse_project_fields:
+                    field_name = reverse_project_fields[cur_field]
+                else:
+                    field_name = "field%s" % index
+                    project_fields[field_name] = cur_field
+                    reverse_project_fields[cur_field] = field_name
+                new_field_name = "$%s" % field_name
+                # _id group
+                if i == 0:
+                    elt[field_name] = new_field_name
+                # collect group
+                else:
+                    elt[field_name] = {'$push': new_field_name}
+                index += 1
+        # Add sum projection if sum_fields are provided
+        if sum_fields:
+            project_fields['_sum'] = {
+                '$add': ['$%s' % field for field in internal_fields[2]]}
+
+        pipeline.append({'$project': project_fields})
+
+        # Group stage
+        group = group_fields[1]
+        group['_id'] = group_fields[0]
+        group['_count'] = {'$sum': '$_sum' if sum_fields else 1}
+        pipeline.append({'$group': group})
+
+        pipeline.append({"$sort": {"_count": 1 if least else -1}})
+
+        if skip is not None:
+            pipeline.append({"$skip": skip})
+        if topnbr is not None:
+            pipeline.append({"$limit": topnbr})
+
+        res = self.db[self.columns[self.column_flow]].aggregate(pipeline,
+                                                                cursor={})
+        for entry in res:
+            # Translate again the collected fields
+            ext_entry = {}
+            for key, value in entry.items():
+                if key in project_fields:
+                    ext_entry[project_fields[key][1:]] = value
+                else:
+                    ext_entry[key] = value
+            # Translate again the aggr fields
+            ext_entry['_id'] = {}
+            for key, value in entry['_id'].items():
+                if key in project_fields:
+                    ext_entry['_id'][project_fields[key][1:]] = value
+                else:
+                    ext_entry['_id'][key] = value
+            # apply internal2ip to addr results
+            for addr_field in ['src_addr', 'dst_addr']:
+                addr0, addr1 = (addr_field + '_0', addr_field + '_1')
+                addr = addr_field[:3] + '.addr'
+                # Apply in aggregate fields
+                if addr0 in ext_entry['_id'] and addr1 in ext_entry['_id']:
+                    ext_entry['_id'][addr] = self.internal2ip(
+                        (ext_entry['_id'].pop(addr0),
+                         ext_entry['_id'].pop(addr1)))
+                # Apply in collected fields
+                if addr0 in ext_entry and addr1 in ext_entry:
+                    ext_entry[addr] = []
```

Comment:
```suggestion
                    ext_entry[addr] = [
                        self.internal2ip(a, b) for a, b in
                        zip(ext_entry[addr0], ext_entry[addr1])
                    ]
```
You have to `from future.builtins import zip`.